cobspec/src/main/java/com/saralein/cobspec/Main.java

```diff
@@ -0,0 +1,78 @@
+package com.saralein.cobspec;
+
+import com.saralein.cobspec.controller.*;
+import com.saralein.cobspec.controller.form.*;
+import com.saralein.cobspec.controller.OptionsController;
+import com.saralein.cobspec.data.FormStore;
+import com.saralein.server.logger.ConnectionLogger;
+import com.saralein.server.logger.Logger;
+import com.saralein.server.protocol.Methods;
+import com.saralein.cobspec.router.RoutesBuilder;
+import com.saralein.server.HttpServer;
+import com.saralein.server.router.Routes;
+import com.saralein.cobspec.validation.ArgsValidation;
+import com.saralein.cobspec.validation.DirectoryValidator;
+import com.saralein.cobspec.validation.PortValidator;
+import com.saralein.cobspec.validation.Validator;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) {
+        String home = System.getProperty("user.dir");
+        Logger logger = new ConnectionLogger(System.out);
+        List<String> validationErrors = runValidationAndReturnErrors(args, home);
+
+        if (validationErrors.isEmpty()) {
+            ArgsParser argsParser = new ArgsParser(args);
+            Integer port = argsParser.parsePort();
+            Path root = argsParser.parseRoot(home);
+            Routes routes = setupRoutes(root);
+            HttpServer.start(port, root, routes, logger);
+        } else {
+            logger.log(String.join("\n", validationErrors));
```

Comment:
This is a bit minor, but it looks like if the validation failed, you would have an exit code of 0, which implies that the process completed successfully.